fastf1/core.py

```diff
@@ -2557,8 +2557,32 @@ def pick_lap(self, lap_number: int) -> "Laps":
         Returns:
             instance of :class:`Laps`
         """
+        warnings.warn(("This method will be deprecated in the next "
+                       "minor release, use :meth:`pick_laps` instead."),
+                      DeprecationWarning)
         return self[self['LapNumber'] == lap_number]
 
+    def pick_laps(self, lap_numbers: Union[int, Iterable[int]]) -> "Laps":
+        """Return all laps of a specific LapNumber or a list of LapNumbers
+        in self
+
+            lap_1 = ff1.pick_laps(1)
+            lap_10_to_20 = ff1.pick_lap(range(10, 21))
+
+        Args:
+            lap_number (int or iterable): int for matching a single
+            lap, a iterable of ints for matching multiple laps
+
+        Returns:
+            instance of :class:`Laps`
+        """
+        if isinstance(lap_numbers, int):
+            return self[self['LapNumber'] == lap_numbers]
+
+        lap_numbers = [i for i in lap_numbers if isinstance(int, i)]
```

Comment:
Not sure if I am understanding you fully but I just changed this line to `lap_numbers = [int(i) for i in lap_numbers]`. I am assuming the behavior you want is casting to int where possible, and letting python raise an error otherwise.

Reply:
No, that's not what I meant. I want to accept value like 2.0 but raise a ValueError for values like 2.54
The code after your change will never raise an error for floats, because Python will accept int(2.54) and return 2 without error.

Also, I don't think you actually need to cast to int. Rather, something like (untested):

```
for i in lap_numbers:
    if isinstance(i, int):
        continue
    elif isinstance(i, float) and i.is_integer():
        continue
    else:
        raise ValueError(f"Invalid value '{i}' in `lap_numbers`")        
```

Reply:
Ok, implemented as you suggested.

> I want to accept value like 2.0 but raise a ValueError for values like 2.54

Also added a similar check for a single float argument.

```
if isinstance(lap_numbers, float):
    if lap_numbers.is_integer():
        return self[self['LapNumber'] == lap_numbers]
    else:
        raise ValueError(f"Invalid value {lap_numbers} for `lap_numbers`")
```